### PR TITLE
Text Classification Taskflow

### PR DESCRIPTION
**Pull request by sijunhe**
### PR types
New features

### Description
Implement the Text Classification Taskflow. Note that this task is different from all other tasks that it has no out-of-box zero-shot capabilities. Instead, it's used as a simple inference pipeline.

### CODE REVIEW

File: paddlenlp/taskflow/task.py [hunk 1]
```diff
@@ -237,8 +239,12 @@ def _get_inference_model(self):
                     fp.write(md5)
                     fp.close()
 
-        inference_model_path = os.path.join(self._task_path, "static",
-                                            "inference")
+        # When the user-entered model path is already a static model, skip to_static conversion
+        if self.is_static_model:
+            inference_model_path = self.model
```
Comment (wawltor) [inline paddlenlp/taskflow/task.py]: 这里可能会有个问题就是用户放入的静态图模型结构或者参数和任务不想管，这里就没有办法check了

File: paddlenlp/taskflow/taskflow.py [hunk 2]
```diff
@@ -523,53 +526,67 @@ class Taskflow(object):
...
+            self.kwargs = kwargs
+            task_class = TASKS[self.task][tag][self.model]['task_class']
+            self.task_instance = task_class(model=self.model,
+                                            task=self.task,
+                                            priority_path=self.priority_path,
+                                            **self.kwargs)
+        # task is inference only task such as text classification taskflow
+        else:
+            self.model = model
+            self.kwargs = kwargs
```
Comment (wawltor) [inline paddlenlp/taskflow/taskflow.py]: 这里是不是要update一下 config_args 到kwargs，虽然这个任务没有config，后续如果要支持层次分类、多标签分类 是不是就有config的信息了

File: paddlenlp/taskflow/text_classification.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        self.is_static_model = is_static_model
+        self._construct_tokenizer(model)
+        if self.is_static_model:
+            self._get_inference_model()
+        else:
+            self._construct_model(model)
+        # self._label_map = {0: 'negative', 1: 'positive'}
+
+    def _construct_input_spec(self):
+        """
```
Comment (wawltor) [inline paddlenlp/taskflow/text_classification.py:102]: 这里 """ 没有对齐 ，麻烦check一下其他的文件

所以这里已经传入的一个静态图文件？

File: paddlenlp/taskflow/text_classification.py [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+                    self.predictor.run()
+                    logits = self.output_handle[0].copy_to_cpu().tolist()
+                    pred_indices = np.argmax(logits, axis=-1)
+                    probs = softmax(logits, axis=-1)
+                    for prob, pred_index in zip(probs, pred_indices):
+                        model_outputs.append({
+                            "label": pred_index,
+                            "score": prob[pred_index]
+                        })
+        else:
```
Comment (wawltor) [inline paddlenlp/taskflow/text_classification.py:175]: 这个分支是不是根本走不进去，是否考虑删除

File: paddlenlp/taskflow/text_classification.py [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+                        model_outputs.append({
+                            "label": pred_index,
+                            "score": prob[pred_index]
+                        })
+        outputs = {}
+        outputs["text"] = inputs["text"]
+        outputs["model_outputs"] = model_outputs
+        return outputs
+
+    def _postprocess(self, inputs: Dict[str, Any]) -> Dict[str, Any]:
```
Comment (wawltor) [inline paddlenlp/taskflow/text_classification.py:191]: 这里有个问题，后续要不要支持其他的分类任务

File: paddlenlp/taskflow/text_classification.py (hunk 4 above)
Comment (sijunhe) [reply paddlenlp/taskflow/text_classification.py:175]: 下面的分支是动态图分支。动态图和静态图推理取决于用户给的是动态还是静态模型

File: paddlenlp/taskflow/text_classification.py (hunk 3 above)
Comment (sijunhe) [reply paddlenlp/taskflow/text_classification.py:102]: taskflow里的`_construct_input_spec`是用来动转静的，但是我们这里其实是不支持动转静的。我们的逻辑是用户给什么模型就用什么模型推理。原因有二：
1. 我们这里的模型结构可以是任何模型，不像别的task一样写死的，所以不好决定这个input_spec
2. 在auto nlp里面，用户输出的模型可能是动态图（例如刚训练完成的模型）有可能是静态图（例如被trainer compress过的模型）

File: paddlenlp/taskflow/text_classification.py (hunk 5 above)
Comment (sijunhe) [reply paddlenlp/taskflow/text_classification.py:191]: 后续可以支持multi-label, hierarchical的分类，可以到时候再加这个功能

File: paddlenlp/taskflow/taskflow.py (hunk 2 above)
Comment (sijunhe) [reply paddlenlp/taskflow/taskflow.py]: good point!

Comment (sijunhe) [reply paddlenlp/taskflow/taskflow.py]: addressed

File: paddlenlp/taskflow/text_classification.py (hunk 5 above)
Comment (sijunhe) [reply paddlenlp/taskflow/text_classification.py:191]: 这里加了个TODO, 可以后续再实现

File: paddlenlp/taskflow/taskflow.py [hunk 6]
```diff
@@ -389,6 +388,16 @@
             "model": "Salesforce/codegen-350M-mono",
         },
     },
+    "text_classification": {
+        "models": {
+            "multi_class": {
+                "task_class": TextClassificationTask,
+            },
```
Comment (wawltor) [inline paddlenlp/taskflow/taskflow.py:409]: 这个也建议加一个flag

**Review by wawltor — APPROVED** [review]
LGTM
